src/cider/nrepl/middleware/reload.clj

```diff
@@ -0,0 +1,49 @@
+(ns cider.nrepl.middleware.reload
+  (:require
+   [clojure.main :refer [repl-caught]]
+   [clojure.string :as str]
+   [haystack.analyzer :as stacktrace.analyzer]
+   [nrepl.middleware.interruptible-eval :refer [*msg*]]
+   [nrepl.middleware.print :as print]
+   [nrepl.misc :refer [response-for]]
+   [nrepl.transport :as transport]
+   [orchard.misc :as misc]))
+
+(defn- user-reload
+  "clj-reload.core/reload from the user project.
+  Must be configured via clj-reload.core/init before being called."
+  []
+  (some-> (symbol "clj-reload.core" "reload")
+          resolve))
+
+(defn respond
+  [{:keys [transport] :as msg} response]
+  (transport/send transport (response-for msg response)))
+
+(defn- refresh-reply
+  [{:keys [::print/print-fn transport session id] :as msg}]
+  (let [{:keys [exec]} (meta session)]
+    (exec id
+          #(try
+             (let [reload (user-reload)]
+               (when-not reload
+                 (throw (ex-info "clj-reload.core/reload not found" {})))
+               (reload (cond-> {:log-fn (fn [& args]
+                                          (respond msg {:progress (str/join " " args)}))}
+                               (:all msg) (assoc :only :all)))
+               (respond msg {:status :ok}))
+             (catch Throwable error
+               (respond msg {:status :error
+                             :error  (stacktrace.analyzer/analyze error print-fn)})
+               (binding [*msg* msg
+                         *err* (print/replying-PrintWriter :err msg {})]
+                 (repl-caught error))))
+
+          #(respond msg {:status :done}))))
+
+(defn handle-reload [handler msg]
+  (case (:op msg)
+    "reload" (refresh-reply msg)
+    "reload-all" (refresh-reply (assoc msg :all true))
```

Comment:
Is there not a clear op?

Reply:
Hm no, I can't see it in https://github.com/tonsky/clj-reload/blob/main/src/clj_reload/core.clj